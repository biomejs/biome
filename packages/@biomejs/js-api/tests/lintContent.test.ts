import { afterEach, beforeEach, describe, expect, it } from "vitest";
import { Biome, Distribution } from "../dist";

describe("Biome WebAssembly lintContent", () => {
	let biome: Biome;
	beforeEach(async () => {
		biome = await Biome.create({
			distribution: Distribution.NODE,
		});
	});

	afterEach(() => {
		biome.shutdown();
	});

	it("should lint content", () => {
		const result = biome.lintContent("if (a == b) {}", {
			filePath: "example.js",
		});

		expect(result.diagnostics).toMatchSnapshot("lint diagnostics");
	});
});
