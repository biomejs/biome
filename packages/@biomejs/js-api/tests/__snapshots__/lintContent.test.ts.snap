// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Biome WebAssembly lintContent > fixFileMode is SafeAndUnsafeFixes > should fix the code 1`] = `
"
	const a = \`foo \${Date.now()} bar\`;
	const b = / {3}/;
	"
`;

exports[`Biome WebAssembly lintContent > fixFileMode is SafeFixes > should fix the SafeFixes only 1`] = `
"
	const a = "foo " + Date.now() + " bar";
	const b = / {3}/;
	"
`;

exports[`Biome WebAssembly lintContent > fixFileMode is undefined/omitted > should not fix the code 1`] = `
"
	const a = "foo " + Date.now() + " bar";
	const b = /   /;
	"
`;
